src/test/resources/com/puppycrawl/tools/checkstyle/grammar/InputRegressionJavaClass2Ast.txt

```diff
@@ -368,3 +368,32 @@ CLASS_DEF -> CLASS_DEF [38:0]
     |       |   `--SEMI -> ; [40:40]
     |       `--RCURLY -> } [40:42]
     `--RCURLY -> } [41:0]
+INTERFACE_DEF -> INTERFACE_DEF [42:0]
+|--MODIFIERS -> MODIFIERS [42:0]
+|--LITERAL_INTERFACE -> interface [42:0]
+|--IDENT -> i1 [42:10]
+`--OBJBLOCK -> OBJBLOCK [42:13]
+    |--LCURLY -> { [42:13]
+    `--RCURLY -> } [42:14]
+CLASS_DEF -> CLASS_DEF [43:0]
+|--MODIFIERS -> MODIFIERS [43:0]
+|--LITERAL_CLASS -> class [43:0]
+|--IDENT -> c6 [43:6]
+|--TYPE_PARAMETERS -> TYPE_PARAMETERS [43:8]
+|   |--GENERIC_START -> < [43:8]
+|   |--TYPE_PARAMETER -> TYPE_PARAMETER [43:9]
+|   |   |--IDENT -> T [43:9]
+|   |   `--TYPE_UPPER_BOUNDS -> extends [43:11]
+|   |       |--IDENT -> c3 [43:19]
+|   |       |--TYPE_ARGUMENTS -> TYPE_ARGUMENTS [43:21]
+|   |       |   |--GENERIC_START -> < [43:21]
```

Comment:
This is probably a different issue, but based on my previous question, it seems to me `c3` and `<T>` should all be under 1 node and not combined with the `TYPE_EXTENSION_AND`.
I would expect something similar to `TYPECAST`:
````
TYPE_UPPER_BOUNDS -> extends [43:11]
   |-- TYPE
   |   |-- IDENT -> c3
....
   |-- TYPE_EXTENSION_AND
   |-- TYPE
   |   |-- IDENT -> i1
````

Reply:
Our AST is not ideal for historical reasons.
I would expect it to be: TYPE_EXTENSION_AND be parent of c3 and i1.
I think this is how all two argument operation are structured.

Let's postpone it to separate issue, if really care.

Reply:
I agree with @rnveach, there is missing `TYPE` node under `TYPE_UPPER_BOUNDS`.
And yes, this should be a separate issue.

>I would expect it to be: TYPE_EXTENSION_AND be parent of c3 and i1.
I think this is how all two argument operation are structured.

@romani What about `extends c3 & i1 & i2 & i3`? In this case it will be
```
--TYPECAST -> (
   |--TYPE_EXTENSION_AND -> &
   |   |--TYPE_EXTENSION_AND -> &
   |   |   |--TYPE_EXTENSION_AND-> &
   |   |   |   |--TYPE
   |   |   |   |   `--IDENT -> c3
   |   |   |   `--TYPE
   |   |   |       `--IDENT -> i1
   |   |   `--TYPE
   |   |       `--IDENT -> i2
   |   `--TYPE
   |       `--IDENT -> i3
   |--RPAREN -> )
```
Why `i2` should be sibling to pair of (`c3`, `i1`)?
For the `BAND` this makes sense. For `TYPECAST` it doesn't.

Reply:
Let's move this discussion in new issue. AST above is not what I meant/expect.
Let's finish this PR.